developing-key4hep-software/Key4hepCMakeGuide.md

```diff
@@ -0,0 +1,119 @@
+# CMake guide for Key4hep software
+
+## Overview
+
+CMake is a tool for building software, which has become the de-facto
+standard outside HEP. In HEP, it is for example used by the ILC/CLIC
+communities and by the LHCb collaboration. For CMS people, CMake is the
+equivalent of scram.
+
+## Quick start into building Key4hep Software
+
+### Set up the environment
+
+The first step is adding the 
+In case you are unsure, it is best to use the default init script, provided on CVMFS for Centos7:
+
+```
+source /cvmfs/sw.hsf.org/key4hep/setup.sh`
+```
+
+Note that mixing setup scripts (from another package, for example) may or may not work as intended - more likely not.
+For any requests to changes in the environment, feel free to contact the software team on the mailing list or any other channels.
+Developers may also look into `spack` to have more fine-grained control over the build dependencies.
+
+
+### Running CMake
+
+* Create a build directory: `mkdir build; cd build`
+* Run CMake in the build directory: `cmake .. `
+* Change any cmake options by rerunning cmake. For example: `cmake .. -DCMAKE_INSTALL_PREFIX=../install`. Tools like ccmake may also be useful: `ccmake ..`
+* Compile the software, using all the cpus available:    ```make -j `getconf _NPROCESSORS_ONLN` ```  
+* Install by running `make install`
+* In case any dependency is changed, most likely you need to remove all the contents of the build folder and rerun cmake and the compilation.
+
+### Using your local installation
+
+In order to run the code you just installed, there are a few environment variables to set up (assuming the installation directory is the working directory):
+
+* `export CMAKE_PREFIX_PATH=$PWD/:$CMAKE_PREFIX_PATH`
+  in order to use this installation as a dependency for other packages
+* `export PATH=$PWD/bin/:$PATH`
+  in order to make any executables available on the command line
+* `export LD_LIBRARY_PATH=$PWD/lib:$PWD/lib64:$LD_LIBRARY_PATH`
+  in order to make libraries available for linking and as for the plugin systems in Gaudi/DD4hep
+* export ROOT_INCLUDE_PATH=$PWD/include:$ROOT_INCLUDE_PATH
+  in case the package builds ROOT dictionaries
+* `export <PACKAGENAME>=$PWD/share/<PACKAGENAME>`
```

Comment:
I would suggest to provide an example here because the first  <PACKAGENAME> is in all caps while the second is simply the package name e.g. `export K4RECCALORIMETER=$PWD/share/k4RecCalorimeter`

One could on top of that rewrite this line as 
`export PACKAGENAME=$PWD/share/packageName`